library/beaker_provisioner.py

```diff
@@ -0,0 +1,455 @@
+        required: true
+    action:
+        description:
+            - Action to perform
+        required: true
+        choices: 'provision', 'release'
+    distro_tree_id:
+        description:
+            - Distro Tree ID
+        default: 71576
+        required: false
+    web_service:
+        description
+            - Web service protocol
+        default: 'rest'
+        choices: 'rest', 'rpc'
+    ca_cert:
+        description:
+            - CA certificate
+        required: false
+'''
+
+LOAN_COMMENT = "Loaned by Ansible's Module For Beaker"
+SYS_FQDN_PATH = '/systems/{fqdn}/'
+SUCCESS_CODE = 200
+WAIT_BETWEEN_PROVISION_CHECKS = 10
+
+
+class BeakerMachine(object):
+    REQ_URL = dict(
+        ALL='',
+        FREE='free/',
+        LOAN='loans/',
+        VIEW='view/',
+        RETURN='loans/+current',
+        RESERVE='reservations/',
+        RELEASE='reservations/+current',
+        PROVISION='installations/',
+        SYSTEMS='systems/',
+    )
+
+    def __init__(self, url, username, password, fqdn, web_service, ca_cert,
+                 disable_warnings=True):
+        """
+        BeakerMachine initializer.
+
+        :param url: Base URL of the Beaker server
+        :param username: Authentication username
+        :param password: Authentication password
+        :param fqdn: Fully qualified domain name of a system
+        :param disable_warnings: Hide unverified HTTPS requests warnings (
+        default = False)
+        """
+        self.base_url = url
+        self.auth = requests.auth.HTTPBasicAuth(username, password)
+        self.fqdn = fqdn
+        self.web_service = web_service
+        self.ca_cert = ca_cert
+        self.disable_warnings = disable_warnings
+        self.changed = False
+        self.session = None
+        self.details = {}
+
+    def create_session(self):
+        """
+        Creates a session and get some details on the machine.
+        """
+        self.session = requests.Session()
+        self.session.auth = self.auth
+
+        # whether verify the SSL certificate or not
+        self.session.verify = self.ca_cert or False
+
+        if self.disable_warnings:
+            requests.packages.urllib3.disable_warnings()
+
+        if self.web_service == 'rpc':
+            self.session.cookies = requests.cookies.cookiejar_from_dict(
+                self._xml_rpc_auth())
+        else:
+            # cookie workaround - consider changing with _get_last_activity()
+            self.list_systems(limit=1)
+
+        self.details = self.get_system_details()
+
+    def get_system_details(self):
+        """
+        Get system's details
+
+        :return: dict containing details about the system
+        """
+        headers = {'Content-Type': 'application/json'}
+        url = urljoin(self.base_url, SYS_FQDN_PATH.format(fqdn=self.fqdn))
+        resp = self.session.get(url, headers=headers)
+
+        assert resp.status_code == SUCCESS_CODE, "Failed to get system's " \
```

Comment:
Consider using requests.codes.ok here instead of new constant

The requests.codes.* can be used all over this module.